src/Nethermind/Nethermind.Synchronization/SnapSync/ProgressTracker.cs

```diff
@@ -471,5 +542,55 @@ public void Dispose()
                 range?.Dispose();
             }
         }
+
+        public interface IStorageRangeLock
+        {
+            IStorageRangeLock Increment(uint value = 1);
```

Comment:
Why increment returns the lock object? My feeling is that it's to satisfy the `ConcurrentDictionary`. I'd keep it simple and make it void.